etcdutl/snapshot/v3_snapshot.go

```diff
@@ -311,6 +335,88 @@ func (s *v3Manager) saveDB() error {
 	return nil
 }
 
+// unsafeGetLatestRevision returns the latest revision without locking the given transaction.
+func (s *v3Manager) unsafeGetLatestRevision(tx backend.BatchTx) (latest revision, err error) {
```

Comment:
Code is read from top to bottom, please move the function to after `markLatestRevCompacted` as it's the least important one.